xarray/backends/rasterio_.py

```diff
@@ -0,0 +1,116 @@
+from collections import OrderedDict
+import numpy as np
+
+try:
+    import rasterio
+except ImportError:
+    rasterio = False
+
+from .. import DataArray
+from ..core.utils import NDArrayMixin, is_scalar
+from ..core import indexing
+
+
+_ERROR_MSG = ('The kind of indexing operation you are trying to do is not '
+              'valid on rasterio files. Try to load your data with ds.load()'
+              'first.')
+
+
+class RasterioArrayWrapper(NDArrayMixin):
+    """A wrapper around rasterio dataset objects"""
+    def __init__(self, riods):
+        self.riods = riods
+        self._shape = self.riods.count, self.riods.height, self.riods.width
+        self._ndims = len(self.shape)
+
+    @property
+    def dtype(self):
+        return np.dtype(self.riods.dtypes[0])
+
+    @property
+    def shape(self):
+        return self._shape
+
+    def __exit__(self, exception_type, exception_value, traceback):
```

Comment:
I don't think this method does anything -- we never use this array as a context manager.